### ISSUE TITLE
[Bug / Perf] VRAM usage increases significantly (+6 GB) when using lower LoKR rank (Rank 4 vs. Rank 8)

### ISSUE THREAD

**Issue by RitschRatsch666** [shootthesound/Fizgig#144]
Running LoKR training with Rank 4 unexpectedly consumes significantly more VRAM (~6 GB extra) compared to Rank 8 under identical settings, causing out-of-memory issues even on high-VRAM hardware (e.g., RTX 5090 at 1 MP resolution).

Because higher ranks usually introduce more parameters, lower ranks are expected to keep memory usage flat or slightly lower. The sharp VRAM spike on lower ranks suggests potential unintended behavior in factor decomposition, padded intermediate tensors, or auto-selected memory paths.

To Reproduce:

 Configure LoKR training with Rank 8 and observe baseline VRAM usage.

 Change only the rank setting to Rank 4 (keeping all other hyperparameters, resolution, and batch size identical).

Observe the spike in peak VRAM consumption (~6 GB increase).

Screenshots / Context:

<img width="1487" height="653" alt="Image" src="https://github.com/user-attachments/assets/10c4139f-8ffb-4329-a188-e759dfde86c7" />

<img width="1487" height="225" alt="Image" src="https://github.com/user-attachments/assets/9393b0d9-f293-473f-8cc4-af384f7a3915" />

Environment:

OS: Linux (CachyOS)

GPU: NVIDIA RTX 5090

**Comment by shootthesound** [comment]
That's the LoKR **factor**, not a rank, and it runs the other way: the big matrix in a LoKR is `out/factor × in/factor`, so halving the factor from 8 to 4 gives you four times the trainable parameters. On Krea 2 that's roughly 175 M → 700 M, and with plain `adamw` holding two fp32 states per parameter it's about 5 GB more optimizer state plus a GB of gradients — the 6 GB you measured. Factor 4 is the bigger, stronger adapter; the hint under the box said "stronger, bigger files" and it now says what that costs.

If you want factor 4 at 1 MP on a 32 GB card, switch Optimizer Type to `adamw8bit` (a quarter of the optimizer state) or drop Target Megapixels to 0.5. Otherwise factor 8 is the one to stay on. Closing as working as intended — re-open if adamw8bit at factor 4 still runs out.
